File: structures-js/structures-api/src/api/idl/decorators/TenantIdDecorator.ts
import {C3Decorator} from '@kinotic/continuum-idl'

/**
 * Denotes the field that will hold the tenant id to use as the Multi Tenant discriminator field
 * When used this will enable optimistic locking for a Structure / Entity
 */
export class TenantIdDecorator extends C3Decorator {

    constructor() {
        super()
        this.type = 'TenantIdDecorator'
    }
}

File: structures-core/src/main/java/org/kinotic/structures/api/domain/idl/decorators/TenantIdDecorator.java
package org.kinotic.structures.api.domain.idl.decorators;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.kinotic.continuum.idl.api.schema.decorators.C3Decorator;
import org.kinotic.continuum.idl.api.schema.decorators.DecoratorTarget;

import java.util.List;

/**
 * Denotes the field that will hold the tenant id to use as the Multi Tenant discriminator field
 * Created By Navíd Mitchell 🤪on 2/12/25
 */
public final class TenantIdDecorator extends C3Decorator {

    @JsonIgnore
    public static final String type = "TenantIdDecorator";

    public TenantIdDecorator() {
        this.targets = List.of(DecoratorTarget.FIELD);
    }
}
